cpp/src/slice2cpp/Gen.cpp

```diff
@@ -2258,55 +2258,15 @@ Slice::Gen::DataDefVisitor::visitClassDefEnd(const ClassDefPtr& p)
 
     // Emit data members. Access visibility may be specified by metadata.
     const DataMemberList dataMembers = p->dataMembers();
-    const bool prot = p->hasMetadata("protected");
-    bool inProtected = false;
-    bool needSp = true;
```

Comment:
What does the formatting look like without `needSp`?
It seems like this field was for emitting a newline before the first data member, more than it was for protected stuff.